checker/src/main/java/org/checkerframework/checker/optional/RevisedOptionalChecker.java

```diff
@@ -0,0 +1,21 @@
+package org.checkerframework.checker.optional;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import org.checkerframework.checker.nonempty.NonEmptyChecker;
+import org.checkerframework.framework.source.AggregateChecker;
+import org.checkerframework.framework.source.SourceChecker;
+
+/** A version of the Optional Checker that runs the NonEmptyChecker as a subchecker. */
```

Comment:
When should this version be used?  Also note in the manual.

Reply:
The idea I had was to replace the existing Optional Checker entirely with the `RevisedOptionalChecker`, since in theory it would be more precise wrt. container types and `Optional`-producing operations.

However, I think making that change (re-naming this to the `OptionalChecker` and deleting the existing one in this PR would be too destructive for this history.

A compromise would be to merge this as-is and then open another PR with the renaming - what are your thoughts?

Reply:
I would prefer that users not run a checker named "RevisedOptionalChecker".  I think that is your plan as well.

I'm OK with doing the renaming in a separate PR.